### PR TITLE
Campaign management

### PR DESCRIPTION
**Pull request by willyyhuang**
Mega feature branch for #31 
Campaign management
Features:
- Campaign index
- Campaign create/edit/delete
- Campaign join
- Campaign details

### CODE REVIEW

File: resources/js/Components/Form/CampaignCreateForm.tsx [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+        <Grid>
+            <Grid justifyContent='flex-start' container spacing={2}>
+                <Grid item xs={12}>
+                    <Typography>{t('campaign.fill-out-fields')}</Typography>
+                </Grid>
+                <StyledGrid item xs={12} md={type === 'Edit' ? 12 : 5}>
+                    <TextField
+                        fullWidth
+                        id='title'
+                        label='Title'
```
Comment (willyyhuang) [inline resources/js/Components/Form/CampaignCreateForm.tsx:77]: not translated

File: resources/js/Components/Form/CampaignCreateForm.tsx [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+                        onChange={(e) => setData('title', e.target.value)}
+                    />
+                    {errors?.title && <ErrorText message={errors?.title} />}
+                </StyledGrid>
+                {type === 'Create' && <StyledGrid item md={2} />}
+                <StyledGrid item xs={12} md={type === 'Edit' ? 12 : 5}>
+                    <Select
+                        hasNoneOption
+                        id='character_id'
+                        label='Assigned Character'
```
Comment (willyyhuang) [inline resources/js/Components/Form/CampaignCreateForm.tsx:89]: not translated

File: resources/js/Components/Form/CampaignCreateForm.tsx [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+                        onChange={(e) =>
+                            setData('character_id', parseInt(e.target.value))
+                        }
+                        options={characters}
+                    />
+                    {errors?.character_id && <ErrorText message={errors?.character_id} />}
+                </StyledGrid>
+                <StyledGrid item xs={12} md={type === 'Edit' ? 12 : 5}>
+                    <Autocomplete
+                        label='Adventure'
```
Comment (willyyhuang) [inline resources/js/Components/Form/CampaignCreateForm.tsx:101]: not translated

File: resources/js/Components/Form/CampaignCreateForm.tsx [hunk 4]
```diff
@@ -0,0 +1,159 @@
...
+                                post(route('campaign.store'))
+                            }
+                            if (type === 'Edit') {
+                                put(route('campaign.update', [editId]))
+                                if (onCloseDrawer) {
+                                    onCloseDrawer()
+                                }
+                            }
+                        }}>
+                        {type === 'Create' ? 'Create' : 'Save'}
```
Comment (willyyhuang) [inline resources/js/Components/Form/CampaignCreateForm.tsx:149]: not translated

File: resources/js/Components/Table/EntryTable/EntryTable.tsx [hunk 5]
```diff
@@ -69,6 +69,18 @@ const EntryTable = ({
             property: 'session',
             title: t('tableColumn.session'),
         },
+        {
+            property: 'levels',
+            title: 'Level',
```
Comment (willyyhuang) [inline resources/js/Components/Table/EntryTable/EntryTable.tsx:74]: not translated

File: resources/js/Components/Table/EntryTable/EntryTable.tsx [hunk 6]
```diff
@@ -69,6 +69,18 @@ const EntryTable = ({
...
+        },
+        {
+            property: 'gp',
+            title: 'GP',
```
Comment (willyyhuang) [inline resources/js/Components/Table/EntryTable/EntryTable.tsx:78]: not translated

File: resources/js/Components/Table/EntryTable/EntryTable.tsx [hunk 7]
```diff
@@ -69,6 +69,18 @@ const EntryTable = ({
...
+        },
+        {
+            property: 'downtime',
+            title: 'Downtime',
```
Comment (willyyhuang) [inline resources/js/Components/Table/EntryTable/EntryTable.tsx:82]: not translated

File: resources/js/Pages/Campaign/Create/CampaignCreate.tsx [hunk 8]
```diff
@@ -0,0 +1,28 @@
...
+type CampaignCreateType = {
+    characters: CharacterData[]
+    adventures: adventureType[]
+}
+
+const CampaignCreate = ({characters, adventures}: CampaignCreateType) => {
+    const theme = getFontTheme('Form', 16)
+
+    return (
+        <ThemeProvider theme={theme}>
```
Comment (willyyhuang) [inline resources/js/Pages/Campaign/Create/CampaignCreate.tsx:17]: pass down language here

File: resources/js/Types/entries-data.ts [hunk 9]
```diff
@@ -27,4 +27,5 @@ export type EntriesData = {
     updated_at: string
     user_id: number
     rating_data: RatingCategoryType | null
+    campaign?: any
```
Comment (willyyhuang) [inline resources/js/Types/entries-data.ts:30]: type this

**Review by willyyhuang — COMMENTED** [review]
@luigibesani there are a bunch of translation missing

**Comment by m-triassi** [comment]
looks like #484 is one of the last things we'd need to address before we can really call this done! :)

**Review by AnnieTran13 — APPROVED** [review]
Looks goods, reported a few minor bugs that can be fixed later.

**Review by JasonGadoury — APPROVED** [review]
Overall works pretty good. Caught a few bugs but nothing too drastic. We can address them in the next sprint. Good work!

**Pull request merged**